File: 2025/day02/Go/part2/day02pt2.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)


func main() {
	ans := 0

	file, err := os.ReadFile("input.txt")
	if err != nil {
		panic(err)
	}
	s := string(file)

	for _, part := range strings.Split(s, ",") {
		bounds := strings.Split(part, "-")

		// convert to int
		start, err := strconv.Atoi(bounds[0])
		if err != nil {
			panic(err)
		}

		// convert to int
		end, err := strconv.Atoi(bounds[1])
		if err != nil {
			panic(err)
		}

		for i := start; i <= end; i++ {
			str := strconv.Itoa(i)
			n := len(str)

			for j := 1; j <= n / 2; j++ {

				pattern := str[:j]
				repeats := n / j

				if strings.Repeat(pattern, repeats) == str {
					ans += i
					break
				}
				
			}
		}

	}
	
	fmt.Println(ans)

}

File: 2025/day02/Python/day02pt2.py
def main():
    ans = 0

    with open("input.txt", "r", encoding="utf-8") as file:
        s = file.read()

    for part in s.split(","):
        start, end = map(int, part.split("-"))

        for i in range(start, end + 1):
            s = str(i)
            n = len(s)

            for j in range(1, len(s) // 2 + 1):

                pattern = s[:j]
                repeats =  n // j

                # check if pattern repeated is the same as s
                if pattern * repeats == s: # if yes invalid
                    ans += int(s)
                    break

    print(ans)

main()